src/psyclone/tests/dynamo0p3_builtins_test.py

```diff
@@ -292,373 +58,392 @@ def test_dynbuiltfactory_str():
     assert "Factory for a call to a Dynamo built-in" in str(factory)
 
 
-def mesh_code_present(code):
-    '''this test checks for the existance of mesh code. This exists for
-    all builtins with dm = True (although it is not actually required!) so
-    each test can call this function'''
-    assert "      USE mesh_mod, ONLY: mesh_type" in code
-    assert "      TYPE(mesh_type), pointer :: mesh => null()" in code
-    output_dm_1 = (
-        "      !\n"
-        "      ! Create a mesh object\n"
-        "      !\n"
-        "      mesh => f1%get_mesh()\n"
-        "      !\n")
-    print output_dm_1
-    assert output_dm_1 in code
+# ------------- Adding (scaled) fields ------------------------------------- #
 
 
-def test_builtin_set_str():
-    ''' Check that the str method of DynSetFieldScalarKern returns the
+def test_X_plus_Y_str():    # Z = X + Y
+    ''' Test that the str method of DynXPlusYKern returns the
     expected string '''
     _, invoke_info = parse(os.path.join(BASE_PATH,
-                                        "15_single_pointwise_invoke.f90"),
+                                        "15.1.1_X_plus_Y_builtin.f90"),
                            api="dynamo0.3")
     for distmem in [False, True]:
         psy = PSyFactory("dynamo0.3",
                          distributed_memory=distmem).create(invoke_info)
         first_invoke = psy.invokes.invoke_list[0]
         kern = first_invoke.schedule.children[0].children[0]
-        assert str(kern) == "Built-in: Set field to a scalar value"
+        assert str(kern) == "Built-in: Add fields"
 
 
-def test_builtin_set():
-    ''' Tests that we generate correct code for a serial builtin
-    set operation with a scalar passed by value'''
+def test_X_plus_Y():
+    ''' Test that the str method of DynXPlusYKern returns the
+    expected string '''
```

Comment:
This test is for more than just the str method.

Reply:
Corrected and merged as requested above.